lib/vector-core/src/sink.rs

```diff
@@ -43,12 +44,27 @@ impl VectorSink {
     /// # Panics
     ///
     /// This function will panic if the self instance is not `VectorSink::Stream`.
-    pub fn into_stream(self) -> Box<dyn StreamSink + Send> {
+    pub fn into_stream(self) -> Box<dyn StreamSink<EventArray> + Send> {
         match self {
             Self::Stream(stream) => stream,
             _ => panic!("Failed type coercion, {:?} is not a Stream", self),
         }
     }
+
+    /// Converts an event sink into a `VectorSink`
+    pub fn from_event_sink(sink: impl Sink<Event, Error = ()> + Send + Unpin + 'static) -> Self {
+        VectorSink::Sink(Box::new(EventSink::new(sink)))
+    }
+
+    /// Converts an event stream into a `VectorSink`
+    ///
+    /// # Panics
+    ///
+    /// TODO
+    pub fn from_event_stream(sink: impl StreamSink<Event> + Send + 'static) -> Self {
+        let sink = Box::new(sink);
+        VectorSink::Stream(Box::new(EventStream { sink }))
+    }
```

Comment:
The naming here feels a little off. The function accepts a sink, not a stream. We do wrap it with `EventStream`, but that's internal and even `EventStream` itself feels like a slightly confusing name given that it's also a sink.

Reply:
Would `from_event_streamsink` be better?